src/components/SQForm/__tests__/SQFormDropdown.test.js

```diff
@@ -0,0 +1,240 @@
+
+const DEFAULT_PROPS = {
+  name: 'dropdownExample',
+  label: 'Dropdown Example'
+};
+
+const DEFAULT_INTITIAL_VALUES = {
+  initialValues: {
+    dropdownExample: ''
+  }
+};
+
+const DEFAULT_OPTIONS = [
+  {label: 'Test 1', value: 1},
+  {label: 'Test 2', value: 2},
+  {label: 'Test 3', value: 3}
+];
+
+const EMPTY_OPTION = '- -';
+
+const renderDropdown = (
+  options = DEFAULT_OPTIONS,
+  fieldProps = DEFAULT_PROPS
+) => {
+  return <SQFormDropdown {...fieldProps}>{options}</SQFormDropdown>;
+};
+
+it('should render list of options', () => {
+  renderSQForm(renderDropdown(), DEFAULT_INTITIAL_VALUES);
+
+  const expandButton = screen.getByRole('button', {name: EMPTY_OPTION});
+
+  userEvent.click(expandButton);
+
+  const optionsList = screen.getByRole('listbox');
+  expect(optionsList).toBeInTheDocument();
+  expect(within(optionsList).getAllByRole('option')).toHaveLength(
+    DEFAULT_OPTIONS.length
+  );
+});
+
+it('should render with initial value', () => {
+  const initialValues = {
+    dropdownExample: 2
+  };
+
+  renderSQForm(renderDropdown(), {initialValues});
+
+  const labelValue = screen.getByRole('button', {name: /test 2/i});
+  expect(labelValue).toHaveTextContent('Test 2');
+});
+
+it('should update when option is selected', () => {
+  const handleChange = jest.fn();
+  const fieldProps = {
+    ...DEFAULT_PROPS,
+    onChange: handleChange
+  };
+
+  const sqFormDropdown = renderDropdown(DEFAULT_OPTIONS, fieldProps);
+
+  renderSQForm(sqFormDropdown, DEFAULT_INTITIAL_VALUES);
+
+  const expandButton = screen.getByRole('button', {name: EMPTY_OPTION});
+  expect(expandButton).toHaveTextContent(EMPTY_OPTION);
+
+  userEvent.click(expandButton);
+  expect(handleChange).toHaveBeenCalledTimes(0);
+
+  const option = screen.getByText(DEFAULT_OPTIONS[0].label);
+  userEvent.click(option);
+
+  expect(handleChange).toHaveBeenCalledTimes(1);
+
+  const labelValue = screen.getByRole('button', {
+    name: DEFAULT_OPTIONS[0].label
+  });
+  expect(labelValue).toHaveTextContent(DEFAULT_OPTIONS[0].label);
+});
+
+it('should disable option', () => {
+  const options = [
+    {label: 'Test 1', value: 1},
+    {label: 'Test 2', value: 2, isDisabled: true},
+    {label: 'Test 3', value: 3}
+  ];
+
+  const fieldProps = {
+    ...DEFAULT_PROPS
+  };
+
+  const sqFormDropdown = renderDropdown(options, fieldProps);
+
+  renderSQForm(sqFormDropdown, DEFAULT_INTITIAL_VALUES);
+
+  const expandButton = screen.getByRole('button', {name: EMPTY_OPTION});
+  userEvent.click(expandButton);
+
+  const optionsList = screen.getByRole('listbox');
+
+  const disabledOption = within(optionsList).getAllByRole('option')[1];
+  expect(disabledOption).toHaveAttribute('aria-disabled', 'true');
+
+  /*
+    This actually still fires the event! So everything after this fails
+    I found this when trying to research why:
+    https://github.com/testing-library/dom-testing-library/issues/92#issuecomment-530524324
+  
+    userEvent.click(disabledOption);
```

Comment:
For some reason it's still letting me click the disabled option.

Reply:
Here's a possible solution
```js
const expandButton = screen.getByRole('button', {name: EMPTY_OPTION});

// Disabled option is skipped
userEvent.type(expandButton, '{tab}{enter}{arrowdown}{arrowdown}{enter}');
const updatedExpandButton = await screen.findByRole('button', {
  name: /test 3/i
});

expect(updatedExpandButton).toBeInTheDocument();
```

Reply:
Some context on why that's async/await:
When we update the value of that component a state change is triggered in Formik. Normally you'd get a warning saying you need to wrap the action that changes the state in `act`, but we can avoid doing that if we `await` the change in state that we're expecting